### ISSUE TITLE
Unable to connect a neutrino backend LND node to a BTCD node with a .onion address

### ISSUE THREAD

**Issue by bitcoinlizard** [lightningnetwork/lnd#4803]
### Background

Unable to connect a neutrino backend LND node to a BTCD node with a .onion address.

### Your environment

* version of `lnd` - 0.11.1-beta
* which operating system (`uname -a` on *Nix) - 5.9.9-arch1-1
* version of `btcd`, `bitcoind`, or other backend - neutrino backend

### Steps to reproduce

Attempt to start LND with the "neutrino.connect" option pointing to a .onion address.  I also have the "tor.active" option enabled and a confirmed working tor configuration.  

### Expected behaviour

LND should start up correctly.

### Actual behaviour

This error is received:
"unable to initialize neutrino backend: unable to create neutrino light client: tor host is unreachable"

### Additional information

I have confirmed that my btcd node that is serving the neutrino data is working correctly and can receive an inbound tor connection.  I have also confirmed that an LND node is able to sync the neutrino block data with this btcd node.  I have confirmed that the LND node in question will work fine with clearnet neutrino nodes.  Finally, I have confirmed that the LND node has a functional tor configuration.  I'm able to connect to other tor LND nodes with the "lncli connect" command.

The issue appears to me to be that LND will not correctly handle the .onion neutrino node.

**Comment by guggero** [comment]
I looked into the code for Neutrino connections. If `tor.active` is true, then the Tor resolver is used to connect to Neutrino backends. Are you sure the `btcd` node is set up correctly and reachable on the `.onion` address? Do you perhaps need to specify the correct port(s) on both the hidden service and then specifying it as the Neutrino backend?

**Comment by bitcoinlizard** [comment]
I'm pretty confident my BTCD node is setup correctly.  Here is the relevant hidden service config:

HiddenServiceDir /var/lib/tor/btcd/
HiddenServiceVersion 2
HiddenServicePort 8333 127.0.0.1:8333

I have connected to this BTCD node via the tor address from a tor only bitcoind node.  Also, prior to switching this BTCD instance to tor I was able to sync an LND node in Neutrino backend mode.

I am going to send you the address of my BTCD tor node to your keybase account if you want to test a connection to it.  I feel like something in the LND code is broken as LND quickly fails to start when given the .onion address for neutrino.connect almost like it is unable to try the connection to a .onion rather than trying to connect and timing out.
